### ISSUE TITLE
Will not display image at the first time init a SVGKImage

### ISSUE THREAD

**Issue by pikacode** [SVGKit/SVGKit#814]
I am using SVGKit+Kingfisher, and this is my code:
![image](https://github.com/SVGKit/SVGKit/assets/12589575/405cc6ef-12b9-4473-b098-6b87424d0cfb)

v1 display nothing but a frame:
![image](https://github.com/SVGKit/SVGKit/assets/12589575/5b794fb5-abf4-426a-8dfa-e868a68f4708)

v2 display the right image:
![image](https://github.com/SVGKit/SVGKit/assets/12589575/d3d1e72e-63ed-4a30-b48e-d74b441e7141)

so I have to do this to fix:
![image](https://github.com/SVGKit/SVGKit/assets/12589575/572a555d-0bc4-4c14-bbfa-eba24a2818b6)

**Comment by adamgit** [comment]
That's strange. What's in the first image? You had no console errors? Is it loading an external resource and fails/gets rejected first time?

**Comment by pikacode** [comment]
I am using a simulator (iPhone 15, iOS 17.2), only got one console error:
```
*** Assertion failure in +[SVGLength pixelsPerInchForCurrentDevice], SVGLength.m:238
```

v1 and v2 is using the same data, this is just a single execution result, not multiple times. And only the first time init a SVGImage after app launched will got this issue. Maybe "something" did not initial immediately in SVGKit?

**Comment by szaboagoston** [comment]
> I am using a simulator (iPhone 15, iOS 17.2), only got one console error:
> 
> ```
> *** Assertion failure in +[SVGLength pixelsPerInchForCurrentDevice], SVGLength.m:238
> ```
> 
> v1 and v2 is using the same data, this is just a single execution result, not multiple times. And only the first time init a SVGImage after app launched will got this issue. Maybe "something" did not initial immediately in SVGKit?

I have the exact same issue.

**Comment by adamgit** [comment]
There was a recent patch to improve that code, something wrong with that commit?

**Comment by szaboagoston** [comment]
> There was a recent patch to improve that code, something wrong with that commit?
> […](#)
> On Thu, 11 Apr 2024 at 10:49, szaboagoston ***@***.***> wrote: I am using a simulator (iPhone 15, iOS 17.2), only got one console error: *** Assertion failure in +[SVGLength pixelsPerInchForCurrentDevice], SVGLength.m:238 v1 and v2 is using the same data, this is just a single execution result, not multiple times. And only the first time init a SVGImage after app launched will got this issue. Maybe "something" did not initial immediately in SVGKit? I have the exact same issue. — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: ***@***.***>

No, I think that code is the solution. If I choose "3.x" branch in SPM, everything works fine as I see now. My fault was that I used the "Up to next Major Version" and that was 3.0.0 which was released in 2021. Sorry for this and thank you for solving the problem!
Btw, could you make a new release from the 3.x branch (e.g. 3.0.1)?

**Comment by xiaowei-dreame** [comment]
I have the same issue.
